File: sistema-penal/src/models/abstract/Pessoa.ts
import type { IEntity } from "../interface/IEntity";

export abstract class Pessoa implements IEntity {
  id?: string;
  nome: string;
  dataNascimento: Date;
  cpf: string;

  constructor(nome: string, dataNascimento: Date, cpf: string, id?: string) {
    this.id = id;
    this.nome = nome;
    this.dataNascimento = dataNascimento;
    this.cpf = cpf;
  }
}

File: api/SistemaPenal/Entities/Abstractions/Pessoa.cs
using System.ComponentModel.DataAnnotations;
using SistemaPenal.Interfaces;

namespace SistemaPenal.Entities.Abstractions;

public abstract class Pessoa : IEntity
{
    public Guid Id { get; set; }
    [Required]
    public string Nome { get; set; } = null!;
    [Required]
    public DateTime DataNascimento { get; set; }
    [Required]
    public string Cpf { get; set; } = null!;
}